### PR TITLE
update git 

### PR DESCRIPTION
**Pull request by SNiTEBoBy**
merge branch and commit git
Fixes #3

### CODE REVIEW

**Review by SNiTEBoBy — COMMENTED** [review]
#3 
Reviews and change
checkout

**Comment by hixio-mh** [comment]
``git checkout -b git-master next
git pull https://github.com/git/git.git master

File: sparse-index.c [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+
+	if (can_convert) {
+		struct cache_entry *se;
+		se = construct_sparse_dir_entry(istate, ct_path, ct);
+
+		istate->cache[num_converted++] = se;
+		return 1;
+	}
+
+	for (i = start; i < end; ) {
```
[the remark this answers is not in the document]
Comment (sniten) [reply sparse-index.c:63]: help

File: t/helper/test-mergesort.c [hunk 2]
```diff
@@ -46,8 +50,362 @@ int cmd__mergesort(int argc, const char **argv)
...
+	if (!argc)
+		return run_tests(ARRAY_SIZE(argv_default), argv_default);
+	printf("%-9s %-16s %8s %8s %8s %8s %8s %s\n",
+	       "distribut", "mode", "n", "m", "get_next", "set_next",
+	       "compare", "verdict");
+	while (argc--) {
+		int i, j, m, n = strtol(*argv++, NULL, 10);
+		for (i = 0; i < ARRAY_SIZE(dist); i++) {
+			for (j = 0; j < ARRAY_SIZE(mode); j++) {
+				for (m = 1; m < 2 * n; m *= 2) {
```
[the remark this answers is not in the document]
Comment (sniten) [reply t/helper/test-mergesort.c:379]: help

File: reftable/dump.c [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	struct reftable_stack *stack = NULL;
+	struct reftable_write_options cfg = { 0 };
+
+	int err = reftable_new_stack(&stack, stackdir, cfg);
+	if (err < 0)
+		goto done;
+
+	err = reftable_stack_compact_all(stack, NULL);
+	if (err < 0)
+		goto done;
```
[the remark this answers is not in the document]
Comment (sniten) [reply reftable/dump.c:40]: help

File: builtin/index-pack.c [hunk 4]
```diff
@@ -364,56 +394,42 @@ static void set_thread_data(struct thread_local *data)
...
+	list_for_each_prev(pos, &done_head) {
+		struct base_data *b = list_entry(pos, struct base_data, list);
+		if (b->retain_data || b == retain)
+			continue;
+		if (b->data) {
+			free_base_data(b);
+			if (base_cache_used <= base_cache_limit)
+				return;
+		}
+	}
```
[the remark this answers is not in the document]
Comment (sniten) [reply builtin/index-pack.c:421]: help

**Comment by hixio-mh** [comment]
gh pr checkout 3

**Pull request closed without merging**